roles/rpm_ostree_rebase/tasks/main.yml

```diff
@@ -1,39 +1,90 @@
   fail:
-    msg: "remote_name or refspec is not defined"
-  when: remote_name is undefined or
-        refspec is undefined
-
-- name: Add remote
-  command: ostree remote add --no-gpg-verify {{ remote_name }} {{ remote_url }}
-  when: remote_url is defined
-
-- name: Rebase
-  command: rpm-ostree rebase {{ remote_name }}:{{ refspec }}
-  register: rebase
-  retries: 5
-  delay: 60
-  until: rebase|success
-  when: commit is undefined
-
-- name: Rebase
-  command: rpm-ostree rebase {{ remote_name }}:{{ refspec }} {{ commit }}
-  register: rebase
-  retries: 5
-  delay: 60
-  until: rebase|success
-  when: commit is defined
+    msg: "refspec is not defined"
+  when: refspec is undefined
 
+# Whee!  Nested blocks!  The primary block checks if there is a remote_name
+# defined.  This could be a 'real' remote or it could be an empty string
+# (but unlikely).  Then, if there is a remote_name and a remote_url, we add
+# the remote config.  If the remote_name is not empty, we try to rebase to the
+# remote:refspec or (remote:refspec commit).  Otherwise, we try to rebase to
+# :refspec (or :refspec commit).
+- block:
+
+  # remote_name is not empty, so we assume it is a real remote
+  - block:
+    - name: Add remote
+      command: ostree remote add --if-not-exists --no-gpg-verify {{ remote_name }} {{ remote_url }}
+      when: remote_url is defined
+
+    - name: Rebase to remote:refspec
+      command: rpm-ostree rebase {{ remote_name }}:{{ refspec }}
+      register: rebase
+      retries: 5
+      delay: 60
+      until: rebase|success
+      when: commit is undefined
+
+    - name: Rebase to commit on remote:refspec
+      command: rpm-ostree rebase {{ remote_name }}:{{ refspec }} {{ commit }}
+      register: rebase
+      retries: 5
+      delay: 60
+      until: rebase|success
+      when: commit is defined
+    when: "{{ remote_name|length }} > 0"
+
+  # remote_name is empty, so we assume it is a local branch
+  - block:
+    - name: Rebase to refspec
+      command: rpm-ostree rebase :{{ refspec }}
+      register: rebase
+      retries: 5
+      delay: 60
+      until: rebase|success
+      when: commit is undefined
+
+    - name: Rebase to commit on refspec
+      command: rpm-ostree rebase :{{ refspec }} {{ commit }}
+      register: rebase
+      retries: 5
+      delay: 60
+      until: rebase|success
+      when: commit is defined
+    when: "{{ remote_name|length }} == 0"
+
+  when:
+    - remote_name is defined
+
+# no remote_name defined, so assume it is a local branch
+- block:
+  - name: Rebase to refspec
+    command: rpm-ostree rebase :{{ refspec }}
+    register: rebase
+    retries: 5
+    delay: 60
+    until: rebase|success
+    when: commit is undefined
+
+  - name: Rebase to commit on refspec
+    command: rpm-ostree rebase :{{ refspec }} {{ commit }}
+    register: rebase
+    retries: 5
+    delay: 60
+    until: rebase|success
+    when: commit is defined
+  when:
+    - remote_name is undefined
```

Comment:
I've been experimenting with something like this in my atomic CLI tests for handling additional options to the base command.  Let me know if this makes sense.  If there are holes in the logic let me know so I can update my atomic CLI tests.

```
# Setup optional arguments and set to empty string if it is not passed in
- name: setup optional arguments
  set_fact:
    ror_refspec: "{{ refspec }}"
    ror_remote_name: "{{ remote_name | default('') }}"
    ror_remote_url: "{{ remote_url | default('') }}"
    ror_commit: "{{ commit | default('') }}"

- name: Add remote
  command: ostree remote add --if-not-exists --no-gpg-verify {{ ror_remote_name }} {{ ror_remote_url }}
  when: 
    - ror_remote_url != ""
    - ror_remote_name != ""

# This should handle the following scenarios for rpm-ostree rebase
#   Note: refspec is required so not mentioned below
#   Note: empty string is a no-op when substituted
# remote supplied & commit id is not supplied -> rpm-ostree rebase remote:refspec 
# remote and commit id is not supplied -> rpm-ostree rebase :refspec
# remote and commit id supplied -> rpm-ostree rebase remote:refspec commit
# remote not supplied, commit supplied -> rpm-ostree :refspec commit
- name: Rebase
  command: rpm-ostree rebase {{ ror_remote_name }}:{{ ror_refspec }} {{ ror_commit }}
  register: rebase
  retries: 5
  delay: 60
  until: rebase|success
```